core/core_isolate.rs

```diff
@@ -353,6 +430,49 @@ impl CoreIsolate {
     let mut state = state_rc.borrow_mut();
     state.op_registry.register(name, op)
   }
+
+  /// Registers a callback on the isolate when the memory limits are approached.
+  /// Use this to prevent V8 from crashing the process when reaching the limit.
+  ///
+  /// Calls the closure with the current heap limit and the initial heap limit.
+  /// The return value of the closure is set as the new limit.
+  pub fn add_near_heap_limit_callback<C>(&mut self, cb: C)
+  where
+    C: FnMut(usize, usize) -> usize + 'static,
+  {
+    let boxed_cb = Box::pin(RefCell::new(cb));
+    let data = boxed_cb.as_ptr() as *mut c_void;
+    self.allocations.near_heap_limit_callback_data =
+      Some((boxed_cb, near_heap_limit_callback::<C>));
+    self
+      .v8_isolate
+      .as_mut()
+      .unwrap()
+      .add_near_heap_limit_callback(near_heap_limit_callback::<C>, data);
+  }
+
+  pub fn remove_near_heap_limit_callback(&mut self, heap_limit: usize) {
```

Comment:
Can you add a test for `remove_near_heap_limit_callback`? 
I have a bit of trouble believing that the user would actually be able to remove a callback, but I might be overly pessimistic.

Reply:
yeah, this is kinda hard. the point here is: without a callback, V8 crashes with a SIGILL which is impossible to catch in a test. that means it's not possible to assert whether the callback was really not called.

I tested it locally though and the test crashed in the expected way.

Reply:
V8 allows for multiple callbacks - unclear why anyone would want that, but it's possible. So if you add 2 callbacks and then remove 1 does it still die with STILL?

Reply:
Ah I see you've limited it to support only one callback.. That's ok.